src/main/java/de/fearnixx/jeak/service/token/TokenConfiguration.java

```diff
@@ -50,18 +53,25 @@ protected boolean populateDefaultConf(IConfigNode root) {
     }
 
     /**
-     * The token path is defined as: ${pluginId} -> ${controllerName} -> ${token}
+     * Check for the {@link TokenScope} of a given token.
      *
-     * @param clazz The class of the controller. It identifies the token.
-     * @return The token as {@link Optional<String>} if the path was resolved. Otherwise an empty {@link Optional<String>}.
+     * @param token The token to find the scopes for.
+     * @return An instance of {@link TokenScope} with the scopes of the token.
      */
-    public Optional<String> readToken(Class<?> clazz) {
-        logger.debug(MessageFormat.format("reading token for {0}", clazz.getName()));
-        return getConfig().getNode(clazz.getAnnotation(RestController.class).pluginId(), extractClassName(clazz), "token").optString();
-    }
-
-    private String extractClassName(Class<?> clazz) {
-        String[] split = clazz.getName().split("[.]");
-        return split[split.length - 1];
+    public TokenScope getTokenScopes(String token) {
+        logger.debug(MessageFormat.format("reading token {0}", token));
+        IConfigNode tokenNode = getConfig().getNode(token);
+        Set<String> tokenScopes = new HashSet<>();
+        if (!tokenNode.isVirtual()) {
+            Optional<List<IConfigNode>> iConfigNodes = tokenNode.optList();
+            iConfigNodes.ifPresent(localIConfigNodes -> tokenScopes.addAll(
+                    localIConfigNodes.stream()
+                            .map(IConfigNode::optString)
+                            .filter(Optional::isPresent)
+                            .map(Optional::get)
+                            .collect(Collectors.toSet()))
```

Comment:
No need to create an additional intermediate set. Just call ``forEach(tokenScopes::add)``. 🙂

Reply:
Thanks for the tip.